src/sbml/Reaction.cpp

```diff
@@ -1990,9 +2003,7 @@ Reaction::removeChildObject(const std::string& elementName, const std::string& i
 
   if (elementName == "kineticLaw")
   {
-    KineticLaw* t = getKineticLaw();
-    if (unsetKineticLaw() == LIBSBML_OPERATION_SUCCESS)
-      return t;
+    unsetKineticLaw(); // already deletes the kineticLaw, so nothing to do after
```

Comment:
surely there should be a return here

Reply:
the return is at the end of the function ... so i didn't see a need to duplicate it here